### PR TITLE
fix: respond 404 correctly

### PR DESCRIPTION
**Pull request by chamini2**
🤦

### CODE REVIEW

**Review by isidentical — COMMENTED** [review]
lets add a test, i guess?

**Review by squat — APPROVED** [review]
Ah nice

File: projects/fal/tests/test_apps.py [hunk 1]
```diff
@@ -325,6 +330,27 @@ def test_app_no_serve_spec_metadata(
...
 
 
+def test_404_response(
+    test_fastapi_app: str, test_app: str, request: pytest.FixtureRequest
+):
+    try:
+        apps.run(test_app, path="/other", arguments={"lhs": 1, "rhs": 2})
+    except fastapi.HTTPException as e:
+        assert e.status_code == 404
+        assert "Path /other not found" in e.detail
```
Comment (isidentical) [inline projects/fal/tests/test_apps.py:340]: `with pytest.raises()`

File: projects/fal/src/fal/api.py [hunk 2]
```diff
@@ -824,7 +825,14 @@ async def not_found_exception_handler(request: Request, exc: HTTPException):
...
             # that the error comes from the app and not our platform.
-            return HTTPException(404, f"Path {request.url.path} not found")
+            try:
+                if exc.detail == "Not Found":
+                    return JSONResponse(
+                        {"detail": f"Path {request.url.path} not found"}, 404
+                    )
+            except Exception:
+                # If it's not a generic 404, just return the original message.
+                return JSONResponse({"detail": exc.detail}, 404)
```
Comment (isidentical) [inline projects/fal/src/fal/api.py:835]: why is there a try/except here? why not just do `if exc.detail == "not found": return x else: return y`?

Comment (chamini2) [reply projects/fal/src/fal/api.py:835]: wanted to be extra careful of exc.detail maybe not being an attribute? but it _should_ be

**Review by isidentical — APPROVED** [review]

**Pull request merged**